src/Management/src/Endpoint/Metrics/IMetricsResponse.cs

```diff
@@ -4,6 +4,8 @@
 
 namespace Steeltoe.Management.Endpoint.Metrics;
 
+#pragma warning disable S4023 // Interfaces should not be empty
 public interface IMetricsResponse
+#pragma warning restore S4023 // Interfaces should not be empty
```

Comment:
How would the JsonConverter know which one to respond with? It depends on what was requested.

Reply:
This is conceptually a union type:

```
type MetricsResponse : SingleMetric | ManyMetric
type SingleMetric: Names
type ManyMetric: Name, Measurements, AvailableTags
```

C# does not support union types, but it can be done like this:
```c#
class SingleMetric
{
	ISet<string> Names
}

class ManyMetric
{
	string Name
	IList<MetricSample> Measurements // note IList instead of List
	IList<MetricTag> AvailableTags   // note IList instead of List
}

class MetricsResponse
{
	SingleMetric? Single
	ManyMetric? Many
}

// Json converter:
if (response.Single != null)
{
	Serialize(response.Single)
}
else
{
	Serialize(response.Many)
}
```
This assumes they are never populated both (they can both be null). If there's no way to distinguish, add a boolean `IsSingle` to `MetricsResponse`.

Reply:
I found a simpler way, I Unioned the results to a MetricsResponse and it is serializing correctly

Reply:
Nice!